theano/gof/utils.py

```diff
@@ -546,28 +546,28 @@ def remove(predicate, coll):
     import hashlib
 
     def hash_from_code(msg):
-        # hashlib.md5() requires an object that supports buffer interface,
+        # hashlib.sha256() requires an object that supports buffer interface,
         # but Python 3 (unicode) strings don't.
         if isinstance(msg, str):
             msg = msg.encode()
         # Python 3 does not like module names that start with
         # a digit.
-        return 'm' + hashlib.md5(msg).hexdigest()
+        return 'm' + hashlib.sha256(msg).hexdigest()
 
 else:
     import hashlib
 
     def hash_from_code(msg):
         try:
-            return hashlib.md5(msg).hexdigest()
+            return hashlib.sha256(msg).hexdigest()
         except TypeError:
             assert isinstance(msg, np.ndarray)
```

Comment:
I think that some combination of numpy/python don't like that we call directly hashlib.sha256(msg) on ndarray. The Python 3 version don't have that. So when we drop python 2.7, this will be removed. I don't want to chance which version needed that to know if it is safe to remove.